src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_x509chain.c

```diff
@@ -169,6 +169,8 @@ static void MergeStatusCodes(CFTypeRef key, CFTypeRef value, void* context)
         *pStatus |= PAL_X509ChainPartialChain;
     else if (CFEqual(keyString, CFSTR("CriticalExtensions")))
         *pStatus |= PAL_X509ChainHasNotSupportedCriticalExtension;
+    else if (CFEqual(keyString, CFSTR("NameConstraints")))
+        *pStatus |= PAL_X509ChainInvalidNameConstraints;
```

Comment:
Open to suggestions on how to improve this. As mentioned, macOS does not give us "what" constraint was not met with the named constraint like the `X509ChainStatusFlags` enum expects, so we need to pick an enum value to represent "some kind of name constraint violation". On Windows / Linux, InvalidNameConstraints means "these constraints are nonsense", not "some kind of name constraint error" since those platforms will use the correct detailed enum value.

Reply:
They're all suppressed by the same verification flag bit (https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/ChainVerifier.cs#L111-L116), so this is ... largely... OK.

We might want to put a sentence in the documentation for the unattainable flags like "This value is not reported on macOS, InvalidNameConstraint is asserted instead."  It's weird because it's documenting enum values for the behavior of an API that produces them, but there's really no legitimate use of the flags API other than for X509Chain, so it seems OK.